lib/Dialect/TTNN/IR/TTNNOpsAttrs.cpp

```diff
@@ -68,6 +65,50 @@ bool TTNNLayoutAttr::hasInterleavedDRAMTensorMemoryLayout() const {
          (getMemLayout().getValue() == TensorMemoryLayout::Interleaved);
 }
 
+// Calculate the logical shape of the shard.
+// Shard is defined as a piece of the tensor that is mapped to a single grid
+// core. It is assumed that the TensorMemoryLayout is sharded.
```

Comment:
Should we assert for this?

Reply:
This is a utility function for calculating shard shape in case the tensor is block sharded.